test/atomics.jl

```diff
@@ -423,6 +423,69 @@ let a = ARefxy(1, -1)
     @test_throws ConcurrencyViolationError @atomicreplace :monotonic :acquire a.x xchg
 end
 
+let a = AtomicMemory{Float64}(undef, 2)
```

Comment:
put a `@testset` around this?

Reply:
Also we definitely want more tests here. Specifically, testing 32 byte inline structs and a `struct` that has a pointer.

Reply:
Also would be good to make sure that the bitsunion case works properly.

Reply:
We also should have some "real" tests of this that (for example) test that multi-threaded algorithms actually converge to the correct results.

Reply:
sure, I can add more tests with e.g. `APadInt` and maybe an index different from 1 ;) I'm not sure though how much more tests are needed.
I see that the primitives (`memoryref*`) are very well tested and `@atomic` macros do nothing more than using them as intended. Adding lots of tests on the higher level doesn't seem to add much value.

I'd suggest a very similar testset for 
1. `AtomicMemory{APadInt}` (non-standard size)
2. `AtomicMemory{Union{Nothing, Int}}` (union type) and
3. `AtomicMemory{Vector{BigInt}}` (non-isbitstype, contains pointer)  

----

What I'd like to test more is the code generated by `@atomic` macro, but I didn't find a straightforward way of comparing e.g. 
```julia
ex = @macroexpand @atomic ...
@test ex == test_ex 
```
without building the `test_ex` by hand. Any tips on this?

As for "real" examples I will soon™ have something, but it might be a bit too involved to be included in the tests here...

Reply:
yeah that makes sense. In that case, I think this might just need docs/news and be ready to merge (although we possibly should merge my doc PR saying "there are no docs" so that that can be backported to 1.11? not fully sure the best approach here).

Reply:
@oscardssmith Sure, merge the minimal docs PR as you please. I will not be responsive during the next week or so, proceed at your pace.

I've added news as todo entry, will do this as the last step here.